ragna/deploy/_ui/resources/upload.js

```diff
@@ -1,9 +1,26 @@
-function upload(files, token, informationsEndpoint, final_callback) {
-  Promise.all(
-    Array.from(files).map((file) => {
-      return uploadFile(file, token, informationsEndpoint);
-    }),
-  ).then(final_callback);
+function upload(files, token, informationEndpoint, final_callback) {
+
+  async function uploadBatches() {
+
+    let index = 0;
+    let uploaded = [];
+
+    const batchSize = 500;
+    const queue = Array.from(files);
+
+    while(index < files.length) {
+      const batch = queue.splice(0, batchSize);
+      await Promise.all(
+        batch.map(file => uploadFile(file, token, informationEndpoint))
+      ).then(results => {
+        uploaded.push(...results);
+      });
+      index += batch.length;
```

Comment:
It seems we are only using the `index` to track the loop iterations. Instead, could we just directly track the length of the `queue`?

```suggestion
    while(queue.length) {
      const batch = queue.splice(0, batchSize);
      await Promise.all(
        batch.map(file => uploadFile(file, token, informationEndpoint))
      ).then(results => {
        uploaded.push(...results);
      });
```

Since `.splice` actually removes elements the loop should again terminate when we pulled the last batch.